src/flint/fmpz_mat.jl

```diff
@@ -1003,6 +1000,34 @@ function lll!(x::ZZMatrix, ctx::LLLContext = LLLContext(0.99, 0.51))
   return x
 end
 
+# for lll_gram, we use the following helpers to deal with the degenerate case.
+
+function _is_definitely_full_rank(x::ZZMatrix)
+  if nrows(x) == 0
+    return x
```

Comment:
```suggestion
    return true
```
Don't you wanna do something like this here? Otherwise, the function is not type stable

Reply:
ah, that should not be here, copy and paste error